sled-hardware/src/underlay.rs

```diff
@@ -44,13 +46,38 @@ pub enum Error {
 
     #[error("Could not enumerate physical links")]
     FindLinks(#[from] FindPhysicalLinkError),
+
+    #[error("Could not set linkprop")]
+    SetLinkprop(#[from] SetLinkpropError),
+
+    #[error("Could not get linkprop")]
+    GetLinkprop(#[from] GetLinkpropError),
 }
 
 /// Convenience function that calls
 /// `ensure_links_have_global_zone_link_local_v6_addresses()` with the links
 /// returned by `find_chelsio_links()`.
 pub fn find_nics() -> Result<Vec<AddrObject>, Error> {
     let underlay_nics = find_chelsio_links()?;
+
+    // Before these links have any consumers (eg. IP interfaces), set the MTU.
+    // If we have previously set the MTU, do not attempt to re-set.
+    const MTU: &str = "1600";
```

Comment:
Based on @rmustacc's comment [here](https://github.com/oxidecomputer/omicron/issues/3086#issuecomment-1544437094), I think we should use 9k as a L2 (data link) MTU. And then set the L3 (vnic) MTU on a case by case basis. Bootstrap vnic MTUs should be 1500 and others can be decided based on data path needs, i.e crucible vnics can be 9k etc. We should be careful to avoid fragmenting where possible. For example, right now the crucible zones have vnics with a 9k mtu over a 9k etherstub that egress through a 1600 physical MTU. This is likely causing performance issues.

If we can do something like modify the vnic create functions in omicron to require an MTU argument so we're not getting unfortunate implicit values, that would be good.

Reply:
done in 4fcd69da261804f4408fd2e6c1bb345f501a70f4 and 5bb49d2b75693b7642d18593818414f5f8d93331

Reply:
In the latter we're passing the MTU in as an option. We should be more strict and require explicitly setting an actual MTU value. I think the option path will lead to situations of just specifying `None` and hoping for the best.

Reply:
96d2729c635e429b4cbe1b744af33b09fab8ec8b changes this to not optional